release/src/Util.hs

```diff
@@ -254,6 +259,26 @@ artifactFiles allArtifacts art@Artifact{..} = do
         [(directory </> scalaSourceJarIn, outDir </> scalaSourceJarOut) | shouldRelease allArtifacts (PlatformDependent False), Just scalaSourceJarIn <- pure mbScalaSourceJarIn] <>
         [(directory </> javadocJarIn, outDir </> javadocJarOut) | shouldRelease allArtifacts (PlatformDependent False), Just javadocJarIn <- pure mbJavadocJarIn]
 
+-- | Given an artifact, produce a list of pairs of an input file and the Maven coordinates
+artifactCoords :: E.MonadThrow m => AllArtifacts -> Artifact PomData -> m [(MavenCoords, Path Rel File)]
+artifactCoords allArtifacts art@Artifact{..} = do
+    let PomData{..} = artMetadata
+    let (directory, name) = splitBazelTarget artTarget
+    let jarClassifier =  if getPlatformDependent artPlatformDependent then Just osName else Nothing
+    directory <- parseRelDir $ unpack directory
+
+    mainArtifactFile <- parseRelFile $ unpack $ mainFileName artReleaseType name
+    pomFile <- parseRelFile (unpack (name <> "_pom.xml"))
+    mbSourceJarFile <- traverse (parseRelFile . unpack) (sourceJarName art)
+    mbScalaSourceJarFile <- traverse (parseRelFile . unpack) (scalaSourceJarName art)
+    mbJavadocJarFile <- traverse (parseRelFile . unpack) (javadocJarName art)
+
+    pure $ [ (MavenCoords { groupId = pomGroupId, artifactId = pomArtifactId, version = pomVersion, classifier = jarClassifier, artifactType = mainExt artReleaseType }, directory </> mainArtifactFile) | shouldRelease allArtifacts artPlatformDependent] <>
```

Comment:
Adding
```haskell
let mavenCoords classifier artifactType =
      MavenCoords { groupId = pomGroupId, artifactId = pomArtifactId, version = pomVersion, classifier, artifactType }
```
would simplify this code quite a bit. Moving the `shouldRelease allArtifacts (PlatformDependent False)` part out of the last 4 list comprehensions might also make it easier to understand.

Reply:
Much cleaner. Thanks